sdks/go/pkg/beam/core/metrics/store.go

```diff
@@ -145,16 +150,73 @@ type ptCounterSet struct {
 	gauges        map[nameHash]*gauge
 }
 
+// Bundle processing state (START_BUNDLE, PROCESS_BUNDLE, FINISH_BUNDLE)
+type BundleProcState string
+
+const (
+	StartBundle   BundleProcState = "START_BUNDLE"
+	ProcessBundle BundleProcState = "PROCESS_BUNDLE"
+	FinishBundle  BundleProcState = "FINISH_BUNDLE"
```

Comment:
While not technically a state, a "TotalBundle" would be a good way to identically track/handle the total metrics.